sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -777,17 +768,65 @@ object GpuCast extends Arm {
     }
   }
 
+  def castStringToDecimal(
+      input: ColumnView,
+      ansiEnabled: Boolean,
+      dt: DecimalType): ColumnVector = {
+    // 1. Sanitize strings to make sure all are floating points
+    // 2. Identify all fixed point values
+    // 3. Cast String to newDt (newDt = dt.scale + 1). Promote precision if needed. This step is
+    //    required so we can round up if needed in the final step
+    // 4. Now cast to newDt to dt (Decimal to Decimal)
+    def getInterimDecimalPromoteIfNeeded(dt: DecimalType): DecimalType = {
+      if (dt.scale + 1 > dt.precision) {
+        // promote if possible or throw
+        if (dt.precision == Decimal.MAX_LONG_DIGITS) {
+          //We don't support Decimal 128
```

Comment:
This comment does not match the checks being done. MAX_LONG_DIGITS, is the maximum precision that a Decimal64 can support. So something with a precision of MAX_LONG_DIGITS is still allowed as a Decimal64.